src/Component/Homepage.js

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect } from 'react';
+import { GoogleLogin } from 'react-google-login';
+import { useNavigate } from 'react-router-dom';
+
+const axios = require('axios');
+const FormData = require('form-data');
+
+function Homepage() {
+  const navigate = useNavigate();
+
+  // Check AccessToken is valid or not
+  async function CheckAccesstoken(accessToken) {
+    const URL = `https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=${accessToken}`;
+    const res = await fetch(URL);
+    return res.ok;
+  }
+
+  // Generate Accesstoken using Refreshtoken
+  function GenerateAccesstoken(refreshToken) {
+    const data = new FormData();
+    data.append('client_id', process.env.REACT_APP_CLIENT_KEY);
+    data.append('client_secret', process.env.REACT_APP_CLIENT_SECRET);
+    data.append('refresh_token', refreshToken);
+    data.append('grant_type', 'refresh_token');
+
+    const config = {
+      method: 'post',
+      url: 'https://oauth2.googleapis.com/token',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      data,
+    };
+
+    axios(config)
+      .then(function (response) {
+        localStorage.setItem('Access_Token', response.access_token);
+        localStorage.setItem('Gen_Date', new Date().getTime());
+      })
+      .catch(function (error) {});
+  }
+
+  // Generate RefreshToken Using AuthCode
+  function GenerateRefreshToken(AuthCode) {
+    const data = new FormData();
+    data.append('code', AuthCode);
+    data.append('client_id', process.env.REACT_APP_CLIENT_KEY);
+    data.append('client_secret', process.env.REACT_APP_CLIENT_SECRET);
+    data.append('grant_type', 'authorization_code');
+    data.append('redirect_uri', 'http://localhost:3000');
+
+    const config = {
+      method: 'post',
+      url: 'https://oauth2.googleapis.com/token',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      data,
+    };
+
+    axios(config)
+      .then(function (response) {
+        localStorage.setItem('Access_Token', response.data.access_token);
+        localStorage.setItem('Refresh_Token', response.data.refresh_token);
+        localStorage.setItem('Gen_Date', new Date().getTime());
```

Comment:
Why are we storing access_token again here?

Reply:
Because In starting we only get AUth Code. When we generate refresh token means we generate both access_token and refresh_token.